lib/assertions.js

```diff
@@ -1107,17 +1107,23 @@ module.exports = function (expect) {
         this.errorMode = 'nested';
         var that = this;
         return subject.then(function (value) {
-            return that.shift(value, 0);
-        }, function (err) {
-            if (typeof nextAssertion === 'string') {
-                that.argsOutput = function (output) {
-                    output.error(nextAssertion);
-                    var rest = that.args.slice(1);
-                    if (rest.length > 0) {
-                        output.sp().appendItems(rest, ', ');
-                    }
-                };
+            if (expect.findTypeOf(nextAssertion).is('expect.it')) {
+                // Force a failing expect.it error message to be property nested instead of replacing the default error message:
```

Comment:
@sunesimonsen Can you come up with an idea that could help avoid this construct?